### PR TITLE
refactor(rest): improve rest versioning and add certificate arguments

### PR DESCRIPTION
**Pull request by tiagolobocastro**
refactor(rest): improve rest versions
  Part of the motivation for this is that we should not mix api versions
   on the same openapi spec.
  This change makes adding api versions easier by wrapping each version
   module with its own openapi wrapper which then generates and exposes
   versioned api specs and versioned swagger-ui's.
  
  While we're here, make use of the openapi base path to clean up the
  openapi paths and allow for easier to read auto generated code.
  Also clean up some redundant type definitions.

refactor(rest): add cmdline certificate arg
  Adds cmdline arguments to use https certificates.
  Adds argument to use dummy certificates (used by test).

### CODE REVIEW

**Review by paulyoong — APPROVED** [review]

File: mbus-api/src/lib.rs [hunk 1]
```diff
@@ -146,8 +146,8 @@ impl FromStr for Channel {
     type Err = strum::ParseError;
 
     fn from_str(source: &str) -> Result<Self, Self::Err> {
-        match &source[0 ..= 2] {
-            "v0/" => {
+        match source.split('/').next() {
+            Some("v0") => {
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply mbus-api/src/lib.rs]: Done

File: rest/service/src/main.rs [hunk 2]
```diff
@@ -60,46 +73,100 @@ fn init_tracing() -> Option<(Tracer, Uninstall)> {
...
+    }
+}
 
-    // dummy certificates
-    let mut config = ServerConfig::new(NoClientAuth::new());
+fn get_certificates() -> anyhow::Result<ServerConfig> {
+    if CliArgs::from_args().dummy_certificates {
+        get_dummy_certificates()
+    } else {
+        // guaranteed to be `Some` by the require_unless attribute
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply rest/service/src/main.rs:108]: Done

File: mayastor-macros/actix/src/lib.rs [hunk 3]
```diff
@@ -18,7 +18,9 @@ impl Method {
         if attr.len() < 2 {
```
Comment (chriswldenyer) [inline mayastor-macros/actix/src/lib.rs:18]: You are doing another attr.remove() below, is this 2 still valid?

Comment (tiagolobocastro) [reply mayastor-macros/actix/src/lib.rs:18]: good catch, should be 3 :+1:

**Review by chriswldenyer — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**